src/types.js

```diff
@@ -113,3 +113,15 @@ export function isEnumValue(enumObj, s) {
   }
   return false;
 }
+
+/**
+ * Checks if the given key is a property in the map.
+ *
+ * @param {T}  obj a map like property.
+ * @param {string}  key
+ * @return {boolean}
+ * @template T
+ */
+export function hasOwn(obj, key) {
+  return Object.prototype.hasOwnProperty.call(obj, key);
```

Comment:
If so, they'd be mild. It's more from my underscore days, where we can't trust that the prototype won't be monkey patched by some bad actor.